index.js

```diff
@@ -61,7 +61,10 @@ RedirectableRequest.prototype.write = function (data, encoding, callback) {
   if (!(typeof data === "string" || typeof data === "object" && ("length" in data))) {
     throw new Error("data should be a string, Buffer or Uint8Array");
   }
-  if (this._requestBodyLength + data.length <= this._options.maxBodyLength) {
+  if (data instanceof Buffer && data.length === 0) {
+    return callback();
```

Comment:
While this seems to fix the problem, I'm still curious as to why the problem occurs. We are passing `data` to `this._currentRequest.write` below. Why doesn't that call the callback (in the case where the buffer is empty)?

Reply:
Hi @RubenVerborgh ! 

This was specifically my concern, I'm not sure why the callback wasn't finishing its job when the buffer is empty.

Reply:
I understand, but I think we should get to the bottom of this. I want to understand why it fails, because if not, we might be introducing other bugs.

Reply:
Completely makes sense! I'll try to really find what's happening and comment back on here if I do find the cause 😄

Reply:
Found it: https://github.com/nodejs/node/blob/v10.7.0/lib/_http_outgoing.js#L608

Reply:
Also documented here:

```
callback <Function> Callback for when this chunk of data is flushed
Returns: <boolean> false if the stream wishes for the calling code to wait for the 'drain' event to be emitted before continuing to write additional data; otherwise true.
```
https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback